metrics/firebase/docs/features/builds_aggregation/01_firestore_builds_aggregation_design.md

```diff
@@ -0,0 +1,176 @@
+# Firestore Builds Aggregation design
+
+## Introduction
+
+For now, to make builds aggregations we should load all builds and process calculations on the client, because Cloud Firestore does not support the aggregation queries.
+
+This solution is okay if a project has not many builds, but over time, the number of builds growth and it will pull along increase the number of reads from the database and leads to heavy load/price implications.
+
+To resolve the described problem, we've investigated a possibility to provide some aggregation calculations using the back-end, provided by the Firebase using the Cloud Functions. See [Firebase Metrics Aggregation](https://github.com/platform-platform/monorepo/blob/master/metrics/firebase/docs/analysis/02_firebase_metrics_aggregation.md) document for more details.
+
+## References
+
+> Link to supporting documentation, GitHub tickets, etc.
+
+- [Github epic: Reduce Firebase usage / document reads](https://github.com/platform-platform/monorepo/issues/1042)
+- [Firebase aggregation](https://github.com/platform-platform/monorepo/blob/master/metrics/firebase/docs/analysis/02_firebase_metrics_aggregation.md)
+- [Firestore Cloud Function using Dart](https://github.com/platform-platform/monorepo/blob/master/metrics/firebase/docs/analysis/01_using_dart_in_the_firebase_cloud_functions.md)
+
+## Design
+
+> Explain and diagram the technical design.
+
+The Firestore builds aggregation implementation requires changes in the Firestore Database and Firestore Cloud Functions. The design describes new collections we should create and a list of security rules we should apply. There is also information about Firestore Cloud Function triggers.
+
+### Firestore
+
+The section contains information about the main purposes of new collections, their document structures with field descriptions, and a set of security rules.
+
+### `build_days`
+
+> Explain the main purpose of the collection.
+
+The first collection we should create is the `build_days`. It holds builds grouped by the `status` and `day`. Each status contains the count of builds, created per day. 
+
+#### Document Structure
+
+> Explain the structure of the documents under this collection.
+
+We should produce a composite document identifier that consists of a project's id and a day this aggregation document belongs to. This identifier we can use to easily update a value of the `build_days` document.
```

Comment:
Would it be great to provide more details about the document id? In which format are we going to store the `day`? Maybe it would be useful to have an example of the document id?